res/controllers/Traktor-Kontrol-S2-MK2-hid-scripts.js

```diff
@@ -0,0 +1,1389 @@
+/****************************************************************/
+/*      Traktor Kontrol S2 MK2 HID controller script            */
+/*      Copyright (C) 2020, Be <be@mixxx.org>                   */
+/*      Copyright (C) 2017, douteiful                           */
+/*      Based on:                                               */
+/*      Traktor Kontrol S4 MK2 HID controller script v1.00      */
+/*      Copyright (C) 2015, the Mixxx Team                      */
+/*      but feel free to tweak this to your heart's content!    */
+/****************************************************************/
+
+// ==== Friendly User Configuration ====
+// The Cue button, when Shift is also held, can have two possible functions:
+// 1. "REWIND": seeks to the very start of the track.
+// 2. "REVERSEROLL": performs a temporary reverse or "censor" effect, where the track
+//    is momentarily played in reverse until the button is released.
+ShiftCueButtonAction = "REWIND";
+
+TraktorS2MK2 = new function() {
+  this.controller = new HIDController();
+
+  // When true, packets will not be sent to the controller.
+  // Used when updating multiple LEDs simultaneously.
+  this.freeze_lights = false;
```

Comment:
Can you further explain what the purpose for this is? Seems to be this is basically a semaphore, but why is it needed exactly?

Reply:
so multiple LED updates can be sent with a single packet rather than sending intermediate packets

Reply:
So iiuc when this flag is set, changes in HID output are being batched together so they can be sent in a single message later?

Reply:
yes

Reply:
Please use camelCase for variable names if you're renaming this anyway.